pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -280,7 +280,10 @@ def pre_dispatch(self):
 
         self.barrier()
 
-    def post_dispatch(self):
+    def post_dispatch(self) -> None:
+        # If the plugin launched subprocesses, clean up the populated environment variable(s)
+        if self.cluster_environment.creates_children:
+            return
         if "WORLD_SIZE" in os.environ:
             del os.environ["WORLD_SIZE"]
```

Comment:
how about we ask the cluster env to clean up?

self.cluster_environment.clean_up / teardown

Reply:
yes totally agreed